src/AuthManager.ts

```diff
@@ -145,29 +145,55 @@ export class AuthManager {
                             code_verifier: codeVerifier,
                         }),
                     })
-                        .then((response) => {
-                            localStorage.removeItem('codeVerifier');
-                            localStorage.removeItem('codeChallenge');
-                            if (response.status !== 200) {
-                                throw new Error('Failed to exchange code for token');
-                            }
-                            return response.json();
-                        })
-                        .then((exchangeJson) => {
-                            localStorage.setItem('access_token', exchangeJson.access_token);
-                            localStorage.setItem('refresh_token', exchangeJson.refresh_token);
-                            resolve();
-                        })
-                        .catch((error) => {
-                            localStorage.removeItem('codeVerifier');
-                            localStorage.removeItem('codeChallenge');
-                            reject(error);
-                        });
+                      .then((response) => {
+                          localStorage.removeItem('codeVerifier');
+                          localStorage.removeItem('codeChallenge');
+                          if (response.status !== 200) {
+                              throw new Error('Failed to exchange code for token');
+                          }
+                          return response.json();
+                      })
+                      .then((exchangeJson) => {
+                          localStorage.setItem('access_token', exchangeJson.access_token);
+                          localStorage.setItem('refresh_token', exchangeJson.refresh_token);
+                          resolve();
+                      })
+                      .catch((error) => {
+                          localStorage.removeItem('codeVerifier');
+                          localStorage.removeItem('codeChallenge');
+                          reject(error);
+                      });
                 }
             } catch (error) {
                 reject(error);
             }
         });
     }
 
+    public static async validateToken(authServer: string, bearerToken: string): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            try {
+                const { data: revokedIds } = await axios.get(`${authServer}public/revoked_ids`);
```

Comment:
maybe the order of checks should be,
- expiry
- signature verification
- then revoked_ids check

so it's like a more steep funnel were less users reach next steps and less resources is spend in the total checks.